libr/core/cmd_print.c

```diff
@@ -4293,8 +4293,15 @@ static int cmd_print(void *data, const char *input) {
 					if (bbsum + 4096 < linear) {
 						eprintf ("Linear size differs too much from the bbsum, please use pdr instead.\n");
 					} else {
-						r_core_cmdf (core, "pD %d @ 0x%08" PFMT64x,
-							f->_size > 0 ? f->_size: r_anal_fcn_realsize (f), f->addr);
+						ut64 at = f->addr;
+						ut64 sz = f->_size > 0 ? f->_size : r_anal_fcn_realsize (f);
+						ut8 *buf = calloc (sz, 1);
+						(void)r_io_read_at (core->io, at, buf, sz);
+						r_core_print_disasm_function (core, f);
+						core->num->value = r_core_print_disasm (core->print, core, at, buf, sz, sz, 0, 1, 0);
+						r_core_print_disasm_function (core, NULL);
```

Comment:
is the r_core_print_disasm_function() really needed? there are only a few places (okay, 6) where r_core_print_disasm() is called ... add an extra arg?

Reply:
this makes for more work if r_core_print_disasm() needs to be reentrant (if ever)